src/history/history.service.ts

```diff
@@ -8,90 +8,101 @@ import { Injectable } from '@nestjs/common';
 import { ConsoleLoggerService } from '../logger/console-logger.service';
 import { HistoryEntryUpdateDto } from './history-entry-update.dto';
 import { HistoryEntryDto } from './history-entry.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { HistoryEntry } from './history-entry.enity';
+import { UsersService } from '../users/users.service';
+import { NotesService } from '../notes/notes.service';
+import { User } from '../users/user.entity';
+import { Note } from '../notes/note.entity';
+import { NotInDBError } from '../errors/errors';
 
 @Injectable()
 export class HistoryService {
-  constructor(private readonly logger: ConsoleLoggerService) {
+  constructor(
+    private readonly logger: ConsoleLoggerService,
+    @InjectRepository(HistoryEntry)
+    private historyEntryRepository: Repository<HistoryEntry>,
+    private usersService: UsersService,
+    private notesService: NotesService,
+  ) {
     this.logger.setContext(HistoryService.name);
   }
 
-  getUserHistory(username: string): HistoryEntryDto[] {
-    //TODO: Use the database
-    this.logger.warn('Using hardcoded data!');
-    return [
-      {
-        metadata: {
-          alias: null,
-          createTime: new Date(),
-          description: 'Very descriptive text.',
-          editedBy: [],
-          id: 'foobar-barfoo',
-          permissions: {
-            owner: {
-              displayName: 'foo',
-              userName: 'fooUser',
-              email: 'foo@example.com',
-              photo: '',
-            },
-            sharedToUsers: [],
-            sharedToGroups: [],
-          },
-          tags: [],
-          title: 'Title!',
-          updateTime: new Date(),
-          updateUser: {
-            displayName: 'foo',
-            userName: 'fooUser',
-            email: 'foo@example.com',
-            photo: '',
-          },
-          viewCount: 42,
-        },
-        pinStatus: false,
+  async getUserHistory(user: User): Promise<HistoryEntry[]> {
+    return await this.historyEntryRepository.find({
+      where: { user: user },
+      relations: ['note'],
+    });
+  }
+
+  private async getEntry(
+    noteIdOrAlias: string,
+    user: User,
+  ): Promise<HistoryEntry> {
+    const note = await this.notesService.getNoteByIdOrAlias(noteIdOrAlias);
+    return await this.getEntryWithNote(note, user);
+  }
+
+  private async getEntryWithNote(
```

Comment:
We use `By` instead of `With` in other places, please rename.
To make the distinction between this function and `getEntry` more apparent, maybe rename `getEntry` to `getEntryByNoteIdOrAlias`?

Reply:
done